platforms/allwinner-d1/build.rs

```diff
@@ -0,0 +1,36 @@
+use std::env;
+use std::fs::File;
+use std::io::Write;
+use std::path::Path;
+
+use d1_config::PlatformConfig;
+use mnemos_config::buildtime::render_project;
+
+fn main() {
+    let out_dir = env::var("OUT_DIR").expect("No out dir");
+    let dest_path = Path::new(&out_dir);
+    let mut f = File::create(dest_path.join("memory.x")).expect("Could not create file");
+
+    f.write_all(include_bytes!("memory.x"))
+        .expect("Could not write file");
+
+    println!("cargo:rustc-link-search={}", dest_path.display());
+    println!("cargo:rerun-if-changed=memory.x");
+    println!("cargo:rerun-if-changed=build.rs");
+    println!("cargo:rustc-link-arg=-Tmemory.x");
+    println!("cargo:rustc-link-arg=-Tlink.x");
+
+    let lichee_rv = cfg!(feature = "lichee-rv");
+    let mq_pro = cfg!(feature = "mq-pro");
+    let beepy = cfg!(feature = "beepy");
+
+    let name = match (lichee_rv, mq_pro, beepy) {
```

Comment:
hmm, what does the "beepy" cfg indicate here? i kind of figured that the "beepy" feature flag would control whether or not the board is attached to a Beepy (e.g. should spawn the i2c_puppet and beepy display drivers), rather than being a separate target entirely?

Reply:
So I was using "one cfg per toml", it's still probably tbd to determine how we set which config to use. It will probably be driven by `just`, but it could be a feature flag or an env var or something.

Right now the options translate to:

* "lichee-rv in a dock"
* "bare mq-pro board"
* "mq-pro in a beepy"

Reply:
okay, yeah, that makes sense for now. this will have to change when we do a beepy shield for the RV, though. then it will be a matrix of `(board, is_beepy)`

Reply:
okay, yeah, that makes sense for now. this will have to change when we do a beepy shield for the RV, though. then it will be a matrix of `(board, is_beepy)`

Reply:
@hawkw I sort of disagree, we don't *really* have a way to do "overlays", so for now, every "configuration" gets its own toml file. So then we would have:

* lichee-rv-dock.toml
* lichee-rv-beepy.toml
* mq-pro-bare.toml
* mq-pro-beepy.toml